dist/src/main/java/io/camunda/application/commons/search/SearchEngineSchemaInitializer.java

```diff
@@ -46,8 +50,52 @@ public void afterPropertiesSet() throws IOException {
                   searchEngineConfiguration,
                   clientAdapter.objectMapper())
               .withMetrics(schemaManagerMetrics);
-      schemaManager.startup();
+      if (!addShutdownHook(executor)) {
+        LOGGER.info("skipping schema initialization, JVM is shutting down");
+        return;
+      }
+      executor.submit(schemaManager::startup).get();
+    } catch (final InterruptedException ie) {
+      LOGGER.warn("Schema initialization task was interrupted");
+      LOGGER.debug("Stack trace:", ie);
```

Comment:
:question: You use this pattern quite a bit - is there a reason you don't want the stack trace normally? :thinking: